bahmnicore-omod/src/main/java/org/bahmni/module/bahmnicore/web/v1_0/search/BahmniConceptSearchHandler.java

```diff
@@ -59,4 +76,25 @@ public PageableResult search(RequestContext context) throws ResponseException {
         }
     }
 
+    /**
+     * Returns list of unique locales based on the context.getParameter("locale") parameter
+     * <strong>Should</strong> return List of unique results for locales: default and specified (or logged in locale if not specified)
+     */
+
+    private List<Locale> getLocales(RequestContext context) {
+        String locale = context.getParameter("locale");
+
+        List<Locale> localeList = new ArrayList<>();
```

Comment:
Little confused .. should it not be that
- If locale is specified - then, return take **only that** in consideration. Basically, if you specify the locale as parameter - then that takes the precedence. 
- if not, then we try to find out the locales - usersession locale and default locale. 
Right now this seem to always return defaultlocale + locale from parameter

Reply:
@angshu : The [API call ](https://docker.standard.mybahmni.in/openmrs/ws/rest/v1/concept?s=byFullySpecifiedName&locale=fr&name=Nutritional+Values&v=bahmni) sending the parameter `locale=fr` and originating from [here](https://github.com/Bahmni/openmrs-module-bahmniapps/blob/master/ui/app/common/concept-set/services/conceptSetServices.js#L6). 
```
params['locale'] = undefined
$bahmniTranslate.use() = 'fr'

so, params['locale'] ='fr'
```

Since, we are passing locale from UI for this end point, I included passed locale and default locale. 
As per your comment, can I remove param `locale` param in the API call [here](https://github.com/Bahmni/openmrs-module-bahmniapps/blob/master/ui/app/common/concept-set/services/conceptSetServices.js#L6) and update the end point locale calculation logic based on above comment ?

Reply:
One way to do this - is from the getConcepts() 
> params['locale'] = params['locale'] || $bahmniTranslate.use();
The above line should not use "$bahmniTranslate.use()" if locale is not specified through param. 
Can you check how many places "conceptSetService.getConcepts()" is called in the UI code?

Reply:
@angshu we are using in few places. 
<img width="1289" alt="image" src="https://github.com/Bahmni/bahmni-core/assets/56538788/4aadcd17-4abb-439a-b058-2e8903ac0bda">

In all the places, we are passing the params `name` and `v` only. 

so, can I remove the line `params['locale'] = params['locale'] || $bahmniTranslate.use();` and update `getLocales(...)` based on supplied locale only or usersession locale and default locale.